subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -506,6 +506,33 @@ int bt_mesh_adv_enable(void)
 	return 0;
 }
 
+int bt_mesh_adv_disable(void)
+{
+	int err;
+	struct k_work_sync sync;
+
+	for (int i = 0; i < ARRAY_SIZE(advs); i++) {
+		k_work_flush_delayable(&advs[i].work, &sync);
+
+		err = bt_le_ext_adv_stop(advs[i].instance);
+		if (err) {
+			LOG_ERR("Failed to stop adv %d", err);
+			return err;
+		}
+
+		adv_sent(advs[i].instance, NULL);
```

Comment:
Minor: please add a comment that `adv_sent` is called to finish a transmission of an adv buffer that was pushed to the host before the advertiser was stopped, but didn't finish.